### PR TITLE
#10193 Document new APIs ContentTypeApi and FieldApi

### PR DESCRIPTION
**Pull request by erickgonzalez**

### CODE REVIEW

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/contenttype/business/FieldApi.java [hunk 1]
```diff
@@ -23,24 +52,91 @@ default FieldApi instance() {
...
 
+	/**
+	 * Removes a field type to the list of field types. Still not implemented.
+	 * 
+	 * @param type Field that wants to be removed.
+	 */
 	void deRegisterFieldType(Field type);
 
+	/**
+	 * Deletes a field in the DB. Also deletes the field variable.
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/FieldApi.java]: change to : "Deletes a field from a content type (as specified in the passed in Field object)"

File: dotCMS/src/main/java/com/dotcms/contenttype/business/FieldApi.java [hunk 2]
```diff
@@ -23,24 +52,91 @@ default FieldApi instance() {
...
+	 * 
+	 * @param field Field that wants to be deleted.
+	 * @throws DotDataException Error occurred when performing the action.
+	 */
 	void delete(Field field) throws DotDataException;
 
+	/**
+	 * Deletes all the fields in the DB related to a Content Type
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/FieldApi.java]: change to: "Deletes all the fields related to the given Content Type"

File: dotCMS/src/main/java/com/dotcms/contenttype/business/FieldApi.java [hunk 3]
```diff
@@ -23,24 +52,91 @@ default FieldApi instance() {
...
+	 * Retrieves a Field given its id
+	 * 
+	 * @param id Id of the field
+	 * @return Field Object that is related to the id.
+	 * @throws DotDataException Error occurred when performing the action.
+	 */
 	Field find(String id) throws DotDataException;
 	
+	/**
+	 * Retrieves a List of Fields based on the Content Type Id (structure_inode column).
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/FieldApi.java]: let's remove "(structure_inode column)."

File: dotCMS/src/main/java/com/dotcms/contenttype/business/FieldApi.java [hunk 4]
```diff
@@ -63,16 +159,51 @@ default FieldApi instance() {
...
+	 * 
+	 * @param type Content Type Id to search on
+	 * @param fieldVar Field Variable to search on
+	 * @return Field Object that met the Field Variable and related to the Content Type
+	 * @throws DotDataException Error occurred when performing the action.
+	 */
+	Field byContentTypeIdAndVar(String id, String fieldVar) throws DotDataException;
+
+    /**
+	 * Deletes the field variable in the DB.
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/FieldApi.java]: "Deletes a field variable from a field (as specified in the passed in FieldVariable object)."

**Review by dsilvam — CHANGES_REQUESTED** [review]

**Review by dsilvam — APPROVED** [review]

**Pull request merged**